core/os/os.cpp

```diff
@@ -494,7 +494,13 @@ int OS::get_power_percent_left() {
 	return -1;
 }
 
+void *OS::get_stack_bottom() const {
+	return _stack_bottom;
+}
+
 OS::OS() {
+	void *volatile stack_bottom;
```

Comment:
`volatile ` before would mean it points to volatile data, but we'd like to prevent the compiler from optimising the pointer variable itself, so the specifier is put on the pointer (this is similar to `const` before vs after). 

No, this is not the very bottom, but is very close and putting it here makes it uniform for all platforms (OS instance is created within the platform-specific main()). It should be enough for any real usage of the pointer.